internal/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -448,21 +456,31 @@ func (c *enrollCmd) daemonReloadWithBackoff(ctx context.Context) error {
 	if err == nil {
 		return nil
 	}
+	if errors.Is(err, context.DeadlineExceeded) ||
```

Comment:
Why are we only considering the context errors here?

Reply:
That's the retry with backoff, so if the context is cancelled or the deadline is exceeded, then we should stop retrying and just return. Any other error we can keep on retrying.

Deadline exceeded and context cancelled are errors that are not related to the daemon reload and because the context is cancelled retrying will always result in the same error.

Reply:
Why is this first call separated out? I guess it was like that before but the way this function is implemented is harder to follow than usual.